examples/frame_prediction/js/entities/entities.js

```diff
@@ -0,0 +1,76 @@
+/************************************************************************************/
+/*                                                                                  */
+/*     Base entity that predicts its position based on the last update              */
+/*                                                                                  */
+/************************************************************************************/
+game.FramePredictionEntity = me.Entity.extend({
+   
+    init: function(x, y, settings) {
+        this._super(me.Entity, "init", [x, y , settings]);
+    },
+    
+    draw: function(renderer) {
+        // Predict the next position to draw at based on velocity
+        var updateInterval = (me.timer.getTime() - me.timer.lastUpdate) / ( (1000/me.sys.updatesPerSecond) );
```

Comment:
Note the interval is calculated based on the time from the last update call, *not* the time from the last requestAnimationFrame. A delta time based on your fps as opposed to the updatesPerSecond is useless for frame prediction between updates.